packages/flutter/lib/src/painting/decoration.dart

```diff
@@ -165,6 +165,8 @@ abstract class Decoration extends Diagnosticable {
   /// omitted if there is no chance that the painter will change (for example,
   /// if it is a [BoxDecoration] with definitely no [DecorationImage]).
   BoxPainter createBoxPainter([ VoidCallback onChanged ]);
+
+  BorderRadius getBorderRadius() => null;
```

Comment:
Not all decorations have a border radius. This will probably have to return a more generic clip Path.